### PR TITLE
Use SPIRE Controller Manager for quickstart demo

### LINKED ISSUE #43265: Update SPIRE quickstart to use SPIRE Controller Manager
**Describe the feature request**

The Istio/SPIRE integration demo utilizes the [deprecated SPIRE Kubernetes Workload Registrar](https://github.com/spiffe/spire/blob/main/support/k8s/k8s-workload-registrar/README.md). There is an existing [issue](https://github.com/spiffe/spire/pull/3853) to remove the k8s-workload-registrar from SPIRE. The [SPIRE Controller Manager](https://github.com/spiffe/spire-controller-manager) is now recommended to facilitate the registration of workloads and federated relationships. 

I propose updating `spire-quickstart.yaml` to utilize the SPIRE Controller Manager and remove references to the SPIRE Kubernetes Workload Registrar in the `README.md`.

Related Istio.io issue to update [Istio/SPIRE integration demo](https://istio.io/latest/docs/ops/integrations/spire/): https://github.com/istio/istio.io/issues/12577

**Affected product area (please put an X in all that apply)**

[ ] Ambient
[x] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Affected features (please put an X in all that apply)**

[ ] Multi Cluster
[ ] Virtual Machine
[ ] Multi Control Plane

### PR DESCRIPTION
**Pull request by jaellio**
**Please provide a description of this PR:**

Updates the SPIRE integration demo files to use the SPIRE Controller Manager for workload registration. The k8s workload registrar for SPIRE has been deprecated. The final version will be [shipped with the v1.5.4 SPIRE release](https://github.com/spiffe/spire/pull/3853).

**Change Highlights:**

`README.md`
- References the Controller Manager instead of the workload registrar
- Adds steps to create a `ClusterSPIFFEID` and label the ingress gateway so it can be registered

`clusterspiffeid.yaml`
- Defines the ClusterSPIFFEID to automatically register workloads with the specified label

`istio-spire-config.yaml`
- Updates the IstioOperator to include an init container and match the [Istio/SPIRE integration documentation](https://github.com/istio/istio.io/pull/12338) on master

`spire-quickstart.yaml`
- Removes resources related to the k8s workload registrar
- Adds SPIRE Controller Manager resources
- Changes are based on the [SPIRE Controller Manager demo](https://github.com/spiffe/spire-controller-manager/tree/main/demo) and the [k8s workload registrar -> controller manager migration guide](https://github.com/spiffe/spire-controller-manager/pull/40)
- Updates the `spire-server` from a StatefulSet to a Deployment for simplification

Resolves #43265 

**Testing:**
- Manually tested spire-quickstart in an AKS cluster
- Manually worked through readme and verified expected outputs

### CODE REVIEW

**Comment by bleggett** [comment]
This will also need a doc update in `istio/istio.io`, AFAICT, to update this page: https://istio.io/latest/docs/ops/integrations/spire/

File: samples/security/spire/clusterspiffeid.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+apiVersion: spire.spiffe.io/v1alpha1
+kind: ClusterSPIFFEID
+metadata:
+  name: example
+spec:
+  spiffeIDTemplate: "spiffe://{{ .TrustDomain }}/ns/{{ .PodMeta.Namespace }}/sa/{{ .PodSpec.ServiceAccountName }}"
+  podSelector:
+    matchLabels:
+      spiffe.io/spiffe-id: "true"
```
Comment (bleggett) [inline samples/security/spire/clusterspiffeid.yaml:9]: Nit, but to make what is happening here clearer, can the matchLabels have both of the labels on the sleep service?

```suggestion
      spiffe.io/spiffe-id: "true"
      app: sleep
```

**Review by bleggett — APPROVED** [review]

**Comment by bleggett** [comment]
These changes look fine to me but need to be merged with https://github.com/istio/istio.io/pull/12680 and should probably be held until that is ready to merge as well.

File: samples/security/spire/clusterspiffeid.yaml (hunk 1 above)
Comment (bleggett) [reply samples/security/spire/clusterspiffeid.yaml:9]: Oh nevermind, you're overloading this for the gateway too.

In that case, can we change this to

```
spiffe.io/spire-workload-ca: "true"
```

File: samples/security/spire/sleep-spire.yaml [hunk 2]
```diff
@@ -47,6 +47,7 @@ spec:
     metadata:
       labels:
         app: sleep
+        spiffe.io/spiffe-id: "true"
```
Comment (bleggett) [inline samples/security/spire/sleep-spire.yaml:50]: See above - would prefer `spiffe.io/spire-managed-identity: "true"` to make it clearer this is specifically saying that SPIRE is attesting and managing the workload identity.

File: samples/security/spire/clusterspiffeid.yaml (hunk 1 above)
Comment (jaellio) [reply samples/security/spire/clusterspiffeid.yaml]: Got it, thanks for explaining! Updated the label to `spiffe.io/spire-managed-identity: "true"`

File: samples/security/spire/sleep-spire.yaml (hunk 2 above)
Comment (jaellio) [reply samples/security/spire/sleep-spire.yaml]: Updated!

**Review by kfaseela — APPROVED** [review]

**Pull request merged**